app/views/filters/_form.html.erb

```diff
@@ -42,9 +42,16 @@
           </span>
       </div>
 
-      <div id="granular_form" style="<%= !f.object.granular? ? 'display:none' : '' %>">
+      <div id="override_taxonomy_form" class="<%= f.object.allows_taxonomies_filtering? ? '' : 'hidden' %>">
+        <%= checkbox_f f, :override?,
+                       :help_inline => _("Filters inherit organizations and locations from their role by default. If override field is enabled, filter can override the set of its organizations and locations. Later role changes will not affect such filter. After disabling override field, role organizations and locations apply again."),
```

Comment:
I wanted to be consistent with existing unlimited checkbox. Also checkbox with popover does not look that great. But alright, I'll hide that into popover. I'm not sure how you want me to address disabled locations/organizations. Should I generate different text based on Settings?? We avoid using word taxonomies in user texts.

Ideally after last experience from other PRs I'd ask for exact text you want me to put there.